src/main/java/com/speaker/service/AccountServiceImpl.java

```diff
@@ -37,18 +36,29 @@ public void init() {
                         Collectors.toMap(Pair::getFirst, this::convertToIds, (city1, city2) -> city1))));
     }
 
+    @Override
+    public List<Account> findAllFriendsByAccountId(int accountId) {
+        return accountRepository.findAllFriendsByAccountId(accountId);
+    }
+
+    @Override
+    public Response addFriend(FriendDTO friendDTO) {
+        Optional<Integer> accountId = convertFirstAndLastNameToId(friendDTO.getAccountFirstLastNames());
+        Optional<Integer> friendId = convertFirstAndLastNameToId(friendDTO.getFriendFirstLastNames());
+        if (nonNull(accountId) && nonNull(friendId) && accountId.isPresent() && friendId.isPresent()) {
```

Comment:
accountId is `Optional`, why need to check on `nonNull`?

Reply:
Since Optional is returned, it is enough to check on isPresent (). Thanks for the comment. Error fixed, correction made.